samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -462,6 +441,14 @@ object SamzaContainer extends Logging {
     else
       null
 
+    val finalTaskFactory = TaskFactoryUtil.finalizeTaskFactory(
+      taskFactory match {
+        case null => TaskFactoryUtil.fromTaskClassConfig(config, appRunner)
+        case _ => taskFactory.asInstanceOf[TaskFactory[_]]
```

Comment:
Discussed offline. Will remove since it does not completely solve the type issue when we classload this factory.